crates/primitives/src/l1.rs

```diff
@@ -116,6 +116,46 @@ impl L1TxProof {
     }
 }
 
+/// Includes `L1BlockManifest` along with scan rules that it is applied to
+#[derive(Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize, Arbitrary)]
+pub struct EpochedL1BlockManifest {
+    /// The actual l1 manifest
+    manifest: L1BlockManifest,
+    /// Epoch, whose scan rules are applied to this manifest
+    epoch: Epoch,
```

Comment:
Doesn't make sense to use the below type for this.  It's always referring to the rules that were derived as of the particular checkpoint, and the checkpoints correspond 1:1 with the epochs they are checkpointing.

Reply:
All boils down to the same point mentioned in other comments and the slack thread.